core/src/main/scala/org/http4s/Message.scala

```diff
@@ -24,27 +22,27 @@ sealed trait Message[F[_]] extends MessageOps[F] { self =>
 
   def body: EntityBody[F]
 
-  final def bodyAsText(implicit defaultCharset: Charset = DefaultCharset): Stream[F, String] =
-    charset.getOrElse(defaultCharset) match {
-      case Charset.`UTF-8` =>
-        // suspect this one is more efficient, though this is superstition
-        body.through(utf8Decode)
-      case cs =>
-        body.through(util.decode(cs))
-    }
-
   def attributes: AttributeMap
 
   protected def change(
+      httpVersion: HttpVersion = httpVersion,
       body: EntityBody[F] = body,
       headers: Headers = headers,
       attributes: AttributeMap = attributes): Self
 
-  override def transformHeaders(f: Headers => Headers)(implicit F: Functor[F]): Self =
-    change(headers = f(headers))
+  def withHttpVersion(httpVersion: HttpVersion): Self =
+    change(httpVersion = httpVersion)
 
-  override def withAttribute[A](key: AttributeKey[A], value: A)(implicit F: Functor[F]): Self =
-    change(attributes = attributes.put(key, value))
+  def withHeaders(headers: Headers): Self =
+    change(headers = headers)
+
+  def withHeaders(headers: Header*): Self =
```

Comment:
Is there a better naming idiom we could adopt here and avoid the overload?  I've grown more hostile to those, but don't have a naming suggestion.

Reply:
I don't have a better idea.